File: Programming Fundamentals with Python/Functions/multiplication_sign.py

# Multiplying
def multiplying(num1, num2, num3):
    result = num1 * num2 * num3
    if result == 0:
        return "zero"
    elif result > 0:
        return "positive"
    else:
        return "negative"

# WITHOUT multiplying
def without_multiplying(num1, num2, num3):
    negative_num_count = 0

    if num1 < 0:
        negative_num_count += 1
    if num2 < 0:
        negative_num_count += 1
    if num3 < 0:
        negative_num_count += 1

    if num1 == 0 or num2 == 0 or num3 == 0:
        return "zero"
    elif negative_num_count > 0:
        return "negative"
    else:
        return "positive"


num1 = int(input())
num2 = int(input())
num3 = int(input())

#print(multiplying(num1, num2, num3))
print(without_multiplying(num1, num2, num3))

File: Programming Fundamentals with Java/Methods/MultiplicationSign.java
import java.util.Scanner;

public class MultiplicationSign 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        int num1 = input.nextInt();
        int num2 = input.nextInt();
        int num3 = input.nextInt();

        if (num1 == 0 || num2 == 0 || num3 == 0)
        {
            System.out.println("zero");
        }
        else
        {
            int negativeCount = 0;
            if (num1 < 0) {negativeCount++;}
            if (num2 < 0) {negativeCount++;}
            if (num3 < 0) {negativeCount++;}

            if (negativeCount % 2 == 0)
            {
                System.out.println("positive");
            } 
            else 
            {
                System.out.println("negative");
            }
        }
    }
}
